git-messenger.el

```diff
@@ -321,12 +321,15 @@ and menus.")
 (defun git-messenger:find-vcs ()
   (let ((longest 0)
         result)
-    (dolist (vcs git-messenger:handled-backends result)
+    (dolist (vcs git-messenger:handled-backends)
       (let* ((dir (assoc-default vcs git-messenger:directory-of-vcs))
              (vcs-root (locate-dominating-file default-directory dir)))
         (when (and vcs-root (> (length vcs-root) longest))
           (setq longest (length vcs-root)
-                result vcs))))))
+                result vcs))))
+    (unless result
+      (error "No supported VCS found"))
```

Comment:
Is this message easy to understand for users ? I suppose like following messages are easier than original one.

- `This directory is not version controlled`
- `This directory is not a git/svn/hg repository`

I'm not good at English. Please ignore my suggestion, if the original message has no problem.

Reply:
I agree my original error message can be confusing for people who do not know what a VCS is.

> This directory is not version controlled

I do not really like this message. This error will show up when the current directory is under some version control systems that git-messenger does not support.

> This directory is not a git/svn/hg repository

This version is better, but it hard-codes the back-ends that git-messenger supports. I would prefer a more general message. What about
`git-messenger failed to find a supported version control repository`

Reply:
Good however package name `git-messenger` is redundant.

Reply:
I am OK with "Failed to find a supported version control repository". Commit force pushed.